### ISSUE TITLE
ERROR in ./node_modules/contentful-sdk-core/dist/es-modules/get-user-agent.js

### ISSUE THREAD

**Issue by ahasall** [contentful/contentful-sdk-core#58]
Hello,

I installed the latest Contentful SDK.

When I build my application I have the following error:
**ERROR in ./node_modules/contentful-sdk-core/dist/es-modules/get-user-agent.js**

When I check the source code of get-user-agent.js if find this:
```js
import { platform, release } from 'os';
```
Knowing that I build for the web, and that the  'os' module if for Node.js apps, I suppose that this import is the reason for the error.

Do you have any idea about this issue ?

Thanks

**Comment by axe312ger** [comment]
See here: https://github.com/contentful/contentful.js/blob/master/webpack.config.js#L49-L51

**Comment by ahasall** [comment]
Okay, thank you. I will create the issue on the Angular CLI repository because I don't have direct access to the Webpack config.

**Comment by axe312ger** [comment]
Argh damn, yeah this is nasty :/ Good luck!

**Comment by ahasall** [comment]
Hello @axe312ger,
It's me again :)

The Angular CLI team believe that librairies that are meant to run inside the browser are [not supposed to use Node API](https://github.com/angular/angular-cli/issues/10681).

I am afraid that they are not willing to provide to override that Webpack config.

Do you plan to add a version of your library that doesn't use those APIs ?

Thanks,
Amadou

**Comment by axe312ger** [comment]
Alright, we will have a look how to avoid importing these node core modules on top level.

**Comment by OliverAssad** [comment]
I have ran into the same problem, this time using **rollup** in a svelte/sapper app. 

``` 
✗ client
node_modules/contentful-sdk-core/dist/es-modules/get-user-agent.js (1:9)
'platform' is not exported by node_modules/rollup-plugin-node-resolve/src/empty.js
1: import { platform, release } from 'os';
            ^
2: import { isNode, getNodeVersion } from './utils';
```

rollup offers different plugins to resolve the issue, one of them is `rollup-plugin-commonjs` with named imports [here](https://github.com/rollup/rollup-plugin-commonjs#custom-named-exports) . However this didn't work either.

**Comment by eatarandom** [comment]
@OliverAssad What was your solve here?

**Comment by larrybotha** [comment]
@eatarandom and @ahasall importing the browser module directly appears to work:

```javascript
  import {createClient} from 'contentful/dist/contentful.browser';
```

**Comment by adamszeptycki** [comment]
@larrybotha thanks for that - works for now.
But would be great to have a real solution to this problem.

**Comment by marcolink** [comment]
We addressed this issue with this [PR](https://github.com/contentful/contentful-sdk-core/pull/198( which got released in version  [v6.8.5](https://github.com/contentful/contentful-sdk-core/releases/tag/v6.8.5). 
The updated dependency is available from [contentful.js@8.5.3](https://github.com/contentful/contentful.js/pull/881#issuecomment-911502093)
